WordPress/gutenberg
# Style Engine: add elements to the frontend

## What?
Migrates elements style generation to the style engine.

This PR affects the editor only. 

The related backend PR took care of the element styles for the frontend site:
- https://github.com/WordPress/gutenberg/pull/40987

Tracking PR:
- https://github.com/WordPress/gutenberg/issues/38167

## Why?
To continue the perilous journey of style generation consolidation!

## How?
~The style engine now checks for an `elements` property, and the loops over each element's styles to generate the style rules.~

I've decided to bring the frontend to the same status as the backend (https://github.com/WordPress/gutenberg/pull/40987).

We'll need to think about how, and if, we want to generate styles using the entire elements object.

## Testing Instructions

I'm using Empty Theme.

Create a new post and add the following block code. Or roll your own!

<details>

<summary>Test block HTML</summary>

```html

<p class="has-link-color"><a href="https://test.com">Test</a> paragraph with a link.</p>



<p class="has-link-color"><a href="https://test.com">Test</a> paragraph with a link.</p>



<h6 class="has-text-color has-link-color" style="color:#65e1cc">Heading with a <a href="https://wordpress.org">link</a></h6>



<div class="wp-block-group has-link-color">
<p><a href="https://wordpress.org">Test</a> paragraph with a link inside a group;</p>
</div>


```

</details>

Check that the editor links appear and behave as you'd expect.


<img width="393" alt="Screen Shot 2022-06-17 at 2 30 15 pm" src="https://user-images.githubusercontent.com/6458278/174225154-96c083a1-68b2-4b0d-af36-058efbe20eb7.png">

Run the tests!

```bash
npm run test-unit packages/style-engine/src/test/index.js
```
